frontend/src/modules/Shared/Shares/ShareEditForm.js

```diff
@@ -0,0 +1,485 @@
+import {
+  Box,
+  Button,
+  CardContent,
+  CircularProgress,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableRow,
+  TextField,
+  Typography
+} from '@mui/material';
+import { Defaults, Pager, ShareStatus } from '../../../design';
+import SendIcon from '@mui/icons-material/Send';
+import React, { useCallback, useEffect, useState } from 'react';
+import {
+  addSharedItem,
+  getShareObject,
+  removeSharedItem,
+  revokeItemsShareObject,
+  submitApproval,
+  updateShareRequestReason
+} from '../../Shares/services';
+import { SET_ERROR } from '../../../globalErrors';
+import { DeleteOutlined } from '@mui/icons-material';
+import PropTypes from 'prop-types';
+import { useNavigate } from 'react-router-dom';
+
+const ItemRow = (props) => {
+  const {
+    share,
+    shareStatus,
+    item,
+    onAction,
+    onLoadingStart,
+    onLoadingEnd,
+    enqueueSnackbar,
+    dispatch,
+    client
+  } = props;
+
+  const whatToDo = () => {
+    if (!item.status) return 'Request';
+    if (
+      item.status === 'Revoke_Succeeded' ||
+      item.status === 'PendingApproval' ||
+      item.status === 'Share_Rejected'
```

Comment:
looking at SM - also should include `Share_Failed` which one should be able to delete from right?